### PR TITLE
Add logger, log authorizations to syslog.

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: code/OAuth.php [hunk 1]
```diff
@@ -152,6 +162,12 @@ public function authorize()
             // (true = approved, false = denied)
             $authRequest->setAuthorizationApproved(true);
 
+            $this->logger->info(sprintf(
+                '%s has been authorized to access client %s',
```
Comment (mateusz) [inline code/OAuth.php]: More appropriately the message should be:
`<person> authorised <client_name> (<client_id)) to access global scope on their behalf`

Comment (halkyon) [reply code/OAuth.php]: should it list the scopes requested? e.g. `<person> authorised <client> to access scopes "read_profile, ...." on their behalf`

File: code/OAuth.php [hunk 2]
```diff
@@ -152,6 +162,12 @@ public function authorize()
             // (true = approved, false = denied)
             $authRequest->setAuthorizationApproved(true);
 
+            $this->logger->info(sprintf(
```
Comment (mateusz) [inline code/OAuth.php:165]: Do we want a test?

Comment (halkyon) [reply code/OAuth.php:165]: any idea how you'd hook a functional test into that? it's outputting to somewhere, maybe to a file or something and then read it off?

File: code/OAuth.php (hunk 1 above)
Comment (mateusz) [reply code/OAuth.php]: Yes, great idea - if you can get them out. We are not really looking at those scopes, but logging what was requested is still good

File: code/OAuth.php (hunk 2 above)
Comment (mateusz) [reply code/OAuth.php:165]: I'd replace logger with a mock, and ensure this->logger->info was called. I'm sure you can do something evil like `Injector::inst()->mublebumble('Logger', Mockery::mock('Logger')`

Comment (halkyon) [reply code/OAuth.php:165]: that sounds doable if it was a unit test, but a functional test fires off an http request, how do you get the mock at that point?! or am I completely overlooking how simple it is?

Comment (mateusz) [reply code/OAuth.php:165]: I don't think it does - it executes director->get or something like that, the routing is done internally. Config overrides should still work. I just don't remember what was the right incantation - probably `Injector::inst()->registerService`

Comment (halkyon) [reply code/OAuth.php:165]: oh okay, cool, yeah overriding the logger is easy, I thought it did an external request and would lose state or something. I'll give it a go

**Pull request merged**